### ISSUE TITLE
[P1] iPEPS optimizer treats complex cotangents as descent gradients and becomes global-phase dependent

### ISSUE THREAD

**Issue by yingjerkao** [tenax-lab/tenax#957]
> 🤖 **AI-generated issue** - written by Codex from the requested iPEPS code review.

The reference-C4v optimizer and the ordinary one-site explicit-AD optimizer pass complex `jax.value_and_grad` cotangents directly into Optax updates. For a real objective of complex parameters, the descent vector needs the conjugate of this JAX gradient. A tiny exact product-state example ascends instead of descending; multiplying the initial tensor by a physically irrelevant global phase changes the final energy.

Sources: [reference gradient](https://github.com/tenax-lab/tenax/blob/42aef1f1a94e1ca8400f6187400aa3613e3e175e/src/tenax/algorithms/ipeps_optimize.py#L935-L953), [reference update](https://github.com/tenax-lab/tenax/blob/42aef1f1a94e1ca8400f6187400aa3613e3e175e/src/tenax/algorithms/ipeps_optimize.py#L994-L1010), [ordinary one-site gradient](https://github.com/tenax-lab/tenax/blob/42aef1f1a94e1ca8400f6187400aa3613e3e175e/src/tenax/algorithms/ipeps_optimize.py#L1692), [ordinary Optax update](https://github.com/tenax-lab/tenax/blob/42aef1f1a94e1ca8400f6187400aa3613e3e175e/src/tenax/algorithms/ipeps_optimize.py#L2084). The optimizer builder uses a plain [Optax Adam chain](https://github.com/tenax-lab/tenax/blob/42aef1f1a94e1ca8400f6187400aa3613e3e175e/src/tenax/algorithms/_ipeps_optimize_shared.py#L121-L140).

## Public-API Reproduction

D=chi=1 with a local Z-field Hamiltonian keeps the calculation in the exact product-state manifold. The gate below is distributed over square-lattice bonds so the energy per site is the local Pauli-Z expectation.

```python
import jax
import jax.numpy as jnp
from tenax import CTMConfig, iPEPSConfig, optimize_gs_ad

z = jnp.diag(jnp.array([1., -1.]))
gate = (jnp.kron(z, jnp.eye(2)) + jnp.kron(jnp.eye(2), z))
gate = gate.reshape(2, 2, 2, 2) / 4
A = jnp.array([1., 0.5j]).reshape(1, 1, 1, 1, 2)
for path in ("reference", "explicit"):
    reference = path == "reference"
    cfg = iPEPSConfig(
        max_bond_dim=1, unit_cell="1x1",
        ctm=CTMConfig(
            chi=1, max_iter=5, min_iter=2,
            ctm_ad_mode="c4v_reference" if reference else None,
        ),
        gs_c4v=reference, gs_implicit_ad=reference,
        gs_explicit_ad_steps=2, gs_explicit_ad_warmup=1,
        gs_optimizer="adam", gs_learning_rate=0.05, gs_num_steps=5,
        gs_conv_criterion="grad_norm", gs_grad_norm_tol=1e-10,
        su_init=False, gs_metric_precond=False,
    )
    for phase in (1., 1j):
        _, _, energy = optimize_gs_ad(gate, phase * A, cfg)
        print(path, phase, float(energy))

v = A.ravel() / jnp.linalg.norm(A)
f = lambda v: jnp.real(jnp.vdot(v, z @ v) / jnp.vdot(v, v))
g = jax.grad(f)(v)
print("initial:", f(v), "raw step:", f(v - 0.05*g),
      "conjugated step:", f(v - 0.05*jnp.conj(g)))
```

Observed energies:

```text
reference 1.0 0.6
reference 1j  0.4721556640281467
explicit  1.0 0.6
explicit  1j  0.4439470012981973
initial: 0.6  raw step: 0.678688524590164  conjugated step: 0.46518282988871223
```

The raw gradient is `[0.71554175, 1.43108351j]`. The unrotated optimizer returns its initial best state despite a valid downhill direction. Both initializations represent the same physical state. These two public optimizer paths were measured; other paths using the same update convention should be audited rather than assumed correct.

## Suggested Fix

Convert complex JAX cotangents to the Euclidean gradients expected by optimizer/preconditioner updates, consistently across pytree leaves. Audit directional derivatives and line-search callbacks so they use the matching convention without double conjugation. Add the exact product-state descent check, global-phase controls, and complex-state regression coverage for Adam, CG, and L-BFGS on supported AD paths.

Reproduced on commit `42aef1f1a94e1ca8400f6187400aa3613e3e175e` using `JAX_PLATFORMS=cpu JAX_ENABLE_X64=1 uv run python`. No repository code was modified for these probes.
